crates/sui-types/src/messages.rs

```diff
@@ -2521,12 +2539,37 @@ impl TransactionEffects {
             unwrapped_object_count: self.unwrapped.len(),
             deleted_object_count: self.deleted.len(),
             wrapped_object_count: self.wrapped.len(),
-            event_count: self.events.len(),
+            event_count: self.events_summary.event_count,
             dependency_count: self.dependencies.len(),
         }
     }
 }
 
+#[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct EventsSummary {
+    pub event_count: usize,
```

Comment:
Why do we need event_count?

Reply:
ya don't really need the count, will remove EventSummary and just use the digest instead